src/main.rs

```diff
@@ -34,6 +46,327 @@ pub mod test_harness;
 
-fn main() {
-    println!("Hello, world!");
+/// A `NetworkBehaviour` that delegates to the `Announce`, `Orderbook`, `Take` and `Comit` (execution) behaviours.
+#[derive(NetworkBehaviour)]
+#[behaviour(out_event = "NetworkEvent", poll_method = "poll")]
+#[allow(missing_debug_implementations)]
+pub struct Nectar {
+    pub comit: Comit,
+
+    // TODO: `Orderbook` and `Take` behaviours to be tied in
+    #[behaviour(ignore)]
+    pub orderbook: Orderbook,
+
+    #[behaviour(ignore)]
+    events: VecDeque<NetworkEvent>,
+}
+
+impl Nectar {
+    pub fn new(orderbook: Orderbook) -> Self {
+        Self {
+            orderbook,
+            comit: Comit::default(),
+            events: VecDeque::new(),
+        }
+    }
+
+    fn poll<BIE>(
+        &mut self,
+        _cx: &mut Context<'_>,
+        _params: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<BIE, NetworkEvent>> {
+        if let Some(event) = self.events.pop_front() {
+            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(event));
+        }
+
+        // We trust in libp2p to poll us.
+        Poll::Pending
+    }
+}
+
+/// Handle events from the network layer.
+///
+/// These are:
+/// - order expired (once orderbook is added)
+/// - order taken (once orderbook is added)
+/// - swap finalized
+impl libp2p::swarm::NetworkBehaviourEventProcess<comit::network::comit::BehaviourOutEvent>
+    for Nectar
+{
+    fn inject_event(&mut self, event: comit::network::comit::BehaviourOutEvent) {
+        self.events.push_back(NetworkEvent::SwapFinalized(
+            event.local_swap_id,
+            event.remote_data,
+        ))
+    }
+}
+
+#[allow(missing_debug_implementations)]
+pub struct Orderbook;
+
+impl Orderbook {
+    fn publish(&mut self, order: Order) {}
+}
+
+// TODO: Replace this placeholder with something meaningful
+// Representation of an Order - implement mapping/replace with orderbook order
+// once we integrate the orderbook.
+#[derive(Debug, PartialEq)]
+pub struct Order {
+    pub expiration_timestamp: DateTime<Utc>,
+    // should at lest have: trading_pair, position, buy/sell amounts
+    // TODO: decision needed how to depict buy/sell amounts (sat:millicent ...)
+}
+
+pub enum NetworkEvent {
+    // When an order expires a new order is published.
+    // New orders are only published after the old one has expired.
+    // We assume that this event is emitted by the orderbook.
+    OrderExpired(Order),
+
+    // The orderbook registered that somebody wants to take a specific order.
+    // We assume that this event is emitted by the orderbook.
+    OrderTakeRequest(Order), // Emitted by orderbook
+
+    // Message from the network that is emitted at the end of the announce protocol when the
+    // swap is finalized.
+    SwapFinalized(LocalSwapId, RemoteData),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Action {
+    PublishNewOrder(Order),
+    StartSwapExecution(LocalSwapId, RemoteData),
+}
+
+// Bundles the state of the application
```

Comment:
The state at which point? I am not sure this is the best representation for the balances, let's see how it all fit together.